timdex_dataset_api/dataset.py

```diff
@@ -41,31 +46,307 @@
+        day: str | None = None,
+    ) -> ds.Dataset:
+        """Lazy filter a pyarrow.dataset.Dataset on a TIMDEXDataset.
+
+        This method will construct a single pyarrow.compute.Expression
+        that is combined from individual equality comparison predicates
+        using the provided filters.
+
+        Args:
+            All args are optional and must be passed in as keyword args.
+
+            Run date columns
+                - run_date (str | date | None, optional)
+                - year (str | None, optional)
+                - month (str | None, optional)
+                - day (str | None, optional)
+
+            If 'run_date' is provided, the 'year', 'month', and 'day' values
+            are parsed from 'run_date'.
+
+            Other columns:
+                - timdex_record_id (str | None, optional)
+                - source (str | None, optional)
+                - run_type (str | None, optional)
+                - run_id (str | None, optional)
+                - action (str | None, optional)
+
+        Raises:
+            DatasetNotLoadedError: Raised if `self.dataset` is None.
+                TIMDEXDataset.load must be called before any filter method calls.
+            ValueError: Raised if provided 'run_date' is an invalid type or
+                cannot be parsed.
+
+        Returns:
+            ds.Dataset: Original pyarrow.dataset.Dataset (if no filters applied)
+                or new pyarrow.dataset.Dataset with applied filters.
+        """
+        if not self.dataset:
+            raise DatasetNotLoadedError
+
+        run_date_date: date | None = None
+
+        # instantiate filters dict
+        filters_dict = dict(
+            zip(
+                TIMDEX_DATASET_FILTER_COLUMNS,
+                [
+                    timdex_record_id,
+                    source,
+                    run_date,
+                    run_type,
+                    run_id,
+                    action,
+                    year,
+                    month,
+                    day,
+                ],
+                strict=False,
+            ),
+        )
+
+        # get filters for partition columns ('run_date' or 'run_date' components)
+        # update filters dict
+        if run_date is not None:
+            if isinstance(run_date, str):
+                run_date_date = strict_date_parse(run_date)
+            elif isinstance(run_date, date):
+                run_date_date = run_date
+            else:
+                raise ValueError(
+                    "Provided 'run_date' value must be a string matching format "
+                    "'%Y-%m-%d' or a datetime.date."
+                )
+            filters_dict.update(
+                {
+                    "run_date": run_date_date,
+                    "year": run_date_date.strftime("%Y"),
+                    "month": run_date_date.strftime("%m"),
+                    "day": run_date_date.strftime("%d"),
+                }
+            )
```

Comment:
Noticed that this block is _very_ similar to this block from `get_partition_prefixes()`:

```python
if run_date:
    if isinstance(run_date, str):
        run_date_date = strict_date_parse(run_date)
    elif isinstance(run_date, date):
        run_date_date = run_date
    else:
        raise ValueError(
            "Provided 'run_date' value must be a string matching format "
            "'%Y-%m-%d' or a datetime.date."
        )
    return (
        f"year={run_date_date.strftime("%Y")}/"
        f"month={run_date_date.strftime("%m")}/"
        f"day={run_date_date.strftime("%d")}"
    )
```

Wonder if a private, helper method like `parse_date_filters()` could perform this work, and return a dictionary with all or some of `run_date, year, month, day` set in a dictionary?

Reply:
A[ new private method `parse_date_filters()`](https://github.com/MITLibraries/timdex-dataset-api/pull/31/commits/0cc15c360be5f984b0461beb5826bdd158df46da#diff-fb40971da01d2dcbf97cdad92e2c3f6656d1eaa201505a2f9074e5d52e452142R327-R347) has been added.